toothpick-runtime/src/main/java/toothpick/configuration/Configuration.java

```diff
@@ -0,0 +1,105 @@
+package toothpick.configuration;
+
+import toothpick.Factory;
+import toothpick.MemberInjector;
+import toothpick.config.Binding;
+
+/**
+ * Strategy pattern that allows to change various behaviors of Toothpick.
+ * The default configuration is {@link #production()} and {@link #reflection()}.
+ * A custom configuration can be created and used by toothpick,
+ * it is even possible to use a composition of the built-in configurations.
+ */
+public class Configuration {
+
+  public static Configuration instance = new Configuration();
+
+  private ReflectionConfiguration reflectionConfiguration = new ReflectionOnConfiguration();
+  private RuntimeCheckConfiguration runtimeCheckConfiguration = new RuntimeCheckOffConfiguration();
+
+  /**
+   * Allows to pass custom configurations.
+   *
+   * @param configuration the configuration to use
+   */
+  public static void setConfiguration(Configuration configuration) {
+    instance = configuration;
+  }
+
+  /**
+   * Performs many runtime checks. This configuration
+   * reduces performance. It should be used only during development.
+   * The checks performed are:
+   * <ul>
+   * <li>cycle detection: check that not 2 classes depend on each other. Note that if of them uses a Lazy instance
+   * of the other or a Producer, then there is no such cycle.</li>
+   * <li>illegal binding detection: check no scope annotated class is used as the target of a binding.</li>
+   * </ul>
+   *
+   * @return a development configuration.
+   */
+  public static Configuration development() {
```

Comment:
actually it is not configuration for development. It is returning development like:
```java
Toothpick.setConfiguration(Configuration.development())
```